### PR TITLE
feat(chat): Modify chat module to implement johnpapa styleguide.

### PR DESCRIPTION
**Pull request by rhutchison**
I know this will further stir the pot about which guidelines the project should follow, but the intention is not to delay progress, and instead push forward.

This is not up for debate - but I will take constructive feedback.

### CODE REVIEW

File: modules/chat/client/controllers/chat.client.controller.js [hunk 1]
```diff
@@ -1,43 +1,55 @@
...
+        Socket.connect();
+      }
+
+      // Add an event listener to the 'chatMessage' event
+      Socket.on('chatMessage', function (message) {
+        vm.messages.unshift(message);
+      });
+
+      // Remove the event listener when the controller instance is destroyed
+      $scope.$on('$destroy', function () {
```
Comment (mleanos) [inline modules/chat/client/controllers/chat.client.controller.js]: can we abstract this out into a service?

Comment (rhutchison) [inline modules/chat/client/controllers/chat.client.controller.js]: That can be a separate PR?

File: modules/chat/client/controllers/chat.client.controller.js [hunk 2]
```diff
@@ -1,43 +1,55 @@
...
-      $location.path('/');
-    }
+  ChatController.$inject = ['$scope', '$state', 'Authentication', 'Socket'];
 
-    // Make sure the Socket is connected
-    if (!Socket.socket) {
-      Socket.connect();
-    }
+  function ChatController($scope, $state, Authentication, Socket) {
+    var vm = this;
```
Comment (ilanbiala) [inline modules/chat/client/controllers/chat.client.controller.js]: Would we rather not do "chat", i.e. more appropriate, less generic names? I think it's useful in the HTML and reads better, but it's up to popular opinion as it won't really affect anything other than variable naming within the controllers.

**Comment by ilanbiala** [comment]
LGTM mostly, @lirantal @codydaig :eyes:?

File: modules/chat/client/controllers/chat.client.controller.js (hunk 2 above)
Comment (codydaig) [inline modules/chat/client/controllers/chat.client.controller.js]: If your referring to vm, I'd rather leave it at vm for consistency.

File: modules/chat/client/controllers/chat.client.controller.js (hunk 1 above)
Comment (codydaig) [inline modules/chat/client/controllers/chat.client.controller.js]: Agreed. That should be a separate PR.

**Comment by codydaig** [comment]
I would like to see scope changed to $scope in the tests. But otherwise, LGTM.

File: modules/chat/client/controllers/chat.client.controller.js (hunk 1 above)
Comment (mleanos) [inline modules/chat/client/controllers/chat.client.controller.js]: Since this is a styling refactor, I thought it would be good to address this as well; if we're strictly adhering to John Papa's style-guide. Might be worthwhile to start the precedent now, with regards to SOC & moving things out into services that shouldn't be in controllers.

It's ok if you want to leave this here though. I don't feel super strong about it :)

File: modules/chat/client/controllers/chat.client.controller.js (hunk 2 above)
Comment (mleanos) [inline modules/chat/client/controllers/chat.client.controller.js]: I think "vm" is best. If you start getting into more specific names (i.e. chat), then you'll end up dancing around that name, when naming other more ambiguous members/properties of a controller. To me it's very clear what "vm" is when looking at the code; it's very accurately named, IMO.

**Comment by mleanos** [comment]
@codydaig What's the reasoning behind changing **scope** to **$scope**? Just to follow suit with the other services? If that's the case, then I'd say we should remove the "$" from the others.

**Comment by trainerbill** [comment]
LGTM, if it helps.  excited to see progress on standards

**Comment by codydaig** [comment]
@mleanos It's to match how they are actually used. And in the code, it's referred too as $scope, so I don't think we should be changing how we reference it in the tests.

**Comment by ilanbiala** [comment]
Yeah definitely change to `$scope`

**Comment by ilanbiala** [comment]
I think this PR is fine, and we can immediately add another PR for the service @mleanos referred to. @codydaig once the $scope change is made and the commit message is reworded, LGTM and you can merge.

**Comment by codydaig** [comment]
@mleanos @ilanbiala @trainerbill  I think that's the last piece holding up #874 as well is the $ in front of variables and maybe one of the controller/service names.

**Comment by mleanos** [comment]
@codydaig This PR is related to #874? Or you mean the changes we want here are similar to what we have left on #874? I'm eager to get that merged. 

Thanks for clarifying about the `$`. That seems right to me too.

**Comment by ilanbiala** [comment]
Similarity of the $.

**Comment by rhutchison** [comment]
@ilanbiala @codydaig good to merge?

**Comment by ilanbiala** [comment]
@rhutchison LGTM, just fix the commit message so it follows these guidelines https://github.com/meanjs/mean/blob/master/CONTRIBUTING.md#commit-message-guidelines

**Comment by mleanos** [comment]
LGTM. I took it for a test drive, and everything looks good, and is working as expected. 

Thanks @rhutchison

**Comment by codydaig** [comment]
LGTM.

Thanks @rhutchison!!!
